docs/pages/CRISP/introduction.mdx

```diff
@@ -27,97 +28,117 @@ coercion. CRISP mitigates these risks through:
-- Verifies if the vote is valid
-
-#### **CRISPPolicy.sol**
-
-Manages voting policies and governance:
-
-- Defines voting rules and constraints
-- Manages voter registration
-
-#### **CRISPChecker.sol**
-
-Additional verification and checking logic:
-
-- Vote format validation
-- Eligibility checking
-- Result verification
-
----
+### Client Application
+
+The [client](https://github.com/gnosisguild/enclave/tree/main/examples/CRISP/client) is a React
+application built with TypeScript that provides a voting interface. It uses the CRISP
+[SDK](/CRISP/introduction#javascript-sdk) and [server](/CRISP/introduction#coordination-server)
+endpoints to deliver the following capabilities:
+
+- Connecting wallets via MetaMask
+- Encrypting votes using FHE encryption before submission
+- Generating Noir zero-knowledge proofs for vote validation
+- Displaying real-time updates on voting status and results
+
+### Coordination Server
+
+The [server](https://github.com/gnosisguild/enclave/tree/main/examples/CRISP/server) is a Rust-based
+coordination service that manages the E3 lifecycle and drives the same SDK from a privileged wallet.
+It acts as a relayer and facilitator, handling blockchain interactions on behalf of clients.
+However, clients can bypass the server entirely and interact directly with the blockchain if they
+prefer.
+
+The server's key responsibilities include:
+
+- Monitoring blockchain events and coordinating protocol progression
+- Collecting encrypted votes from smart contracts
+- Triggering FHE computations after voting rounds are closed
+- Publishing computation results back to the blockchain
+- Providing a RESTful API for client interactions
+
+### FHE Program
+
+The [FHE program](https://github.com/gnosisguild/enclave/tree/main/examples/CRISP/program) is a Rust
+guest program that runs inside the RISC Zero zkVM to perform homomorphic addition on encrypted votes
+(BFV ciphertexts). It computes the encrypted tally without decrypting individual votes, ensuring
+voter privacy throughout the process.
+
+The program decodes BFV parameters and ciphertexts, then sums all encrypted votes homomorphically to
+produce a single encrypted ciphertext containing the total. RISC Zero generates zero-knowledge
+proofs that verify the computation was executed correctly, enabling trustless verification of the
+tally results.
+
+### Zero-knowledge Circuits
+
+[Noir circuits](https://github.com/gnosisguild/enclave/tree/main/examples/CRISP/circuits) provide
+zero-knowledge proofs for vote validation before votes are accepted into the system. These circuits
+verify that votes are correctly encrypted using BFV fully homomorphic encryption under a valid
+public key, check that the voter's address is included in the eligibility Merkle tree, and ensure
+votes conform to the expected structure and values. The circuits also support mask votes, allowing
+anyone to submit zero votes to mask slot activity and prevent receipt-sharing attacks.
```

Comment:
maybe we can simplify the terms here, instead of receipt-sharing attacks, using collusion and coercion. And removing the word "prevent" -> reduces the risk